lib/asyncapiSchemaFormatParser.js

```diff
@@ -11,10 +14,16 @@ module.exports = {
 /**
  * @private
  */
-async function parse({ message, originalAsyncAPIDocument, fileFormat, parsedAsyncAPIDocument, pathToPayload }) {
+async function parse({ schemaFormat, message, defaultSchemaFormat, originalAsyncAPIDocument, fileFormat, parsedAsyncAPIDocument, pathToPayload }) {
+  // setting original schema format required before returning because of no payload
+  message[String(xParserOriginalSchemaFormat)] = schemaFormat || defaultSchemaFormat;
+  
   const payload = message.payload;
   if (!payload) return;
 
+  // considered save here because default parser handles JSON Schema payloads
+  message[String(xParserOriginalPayload)] = JSON.parse(JSON.stringify(payload));
```

Comment:
what about 👇🏼 

```suggestion
  message[String(xParserOriginalPayload)] = Object.assign({}, payload);
```

Reply:
Object.assign creates a new object but the copied properties are still shallow copies. so the payload object itself will not get a `x-parser-schema-id` applied, but for instance a "name" property within the payload will get it applied.

Reply:
you are right.. 🤔 
we cannot leave `JSON.stringify` as it won't work with circular references in schemas

🤔

Reply:
Unfortunately I am not very deep into the node world (I guess you have noticed already 😄 ). So I would need to let you make the call on what to do best.

Considering that [lodash.clonedeep](https://www.npmjs.com/package/lodash.clonedeep) would be available as individual dependency and looking [behind the curtains](https://github.com/lodash/lodash/blob/2f79053d7bc7c9c9561a30dda202b3dcd2b72b90/.internal/baseClone.js#L157) on what would need to be done in an own helper function, I am not sure if manually implementing it is a good idea. On the other hand, I also cannot judge the impact of adding a new dependency to this lib.

Reply:
lodash is just 3.3kb gziped which is fine, I'm more concerned about performance again. We add something that is not that much needed but will add performance issues to large files again 🤔 

I have doubts if this is the way to go. If we should not fall back to the initial idea about adding an additional field that we can use to check if the message was processed or not. Sorry, I just did not take into consideration this issue with copying objects

Reply:
no worries, got you. and yeah, since the payload is not actually transformed, the `x-parser-original-payload` would always include the same structure except the added `x-parser` attributes. not sure if there is any use for this except consistency with custom parsers regarding the `x-parser-original-payload` being there?

Reply:
yeah, the use case if for those that have different formats of schemas, like avro for example, so after parsing the document they can still access the content before conversion in case they want to include it in docs or prefer to read original schema, for example in runtime validations. 

so, again back to `what could be the name of the extension that holds information if the message was already processed or not` 😄